src/khoj/routers/api_agents.py

```diff
@@ -0,0 +1,43 @@
+import json
+import logging
+
+from fastapi import APIRouter, Request
+from fastapi.requests import Request
+from fastapi.responses import Response
+
+from khoj.database.adapters import AgentAdapters
+from khoj.database.models import KhojUser
+from khoj.routers.helpers import CommonQueryParams
+
+# Initialize Router
+logger = logging.getLogger(__name__)
+
+
+api_agents = APIRouter()
+
+
+@api_agents.get("/", response_class=Response)
+async def all_agents(
+    request: Request,
+    common: CommonQueryParams,
+) -> Response:
+    user: KhojUser = request.user.object if request.user.is_authenticated else None
+    agents = await AgentAdapters.aget_all_accessible_agents(user)
+    agents_packet = list()
+    for agent in agents:
+        agents_packet.append(
+            {
+                "slug": agent.slug,
+                "avatar": agent.avatar,
+                "name": agent.name,
+                "tuning": agent.tuning,
```

Comment:
nit: s/tuning/personality/?

```suggestion
                "personality": agent.personality,
```